keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlUnhappyVetoRepository.kt

```diff
@@ -60,21 +59,14 @@ class SqlUnhappyVetoRepository(
     }
   }
 
-  override fun getOrCreateVetoStatus(resourceId: String, application: String, wait: Duration?): UnhappyVetoStatus {
-    val recheckTime: LocalDateTime? = sqlRetry.withRetry(READ) {
+  override fun getRecheckTime(resourceId: String): Instant? {
+    return sqlRetry.withRetry(READ) {
       jooq
         .select(UNHAPPY_VETO.RECHECK_TIME)
         .from(UNHAPPY_VETO)
         .where(UNHAPPY_VETO.RESOURCE_ID.eq(resourceId))
-        .fetchOne()
-        ?.value1()
-    }
-
-    return if (recheckTime == null || recheckTime > clock.timestamp()) {
-      markUnhappyForWaitingTime(resourceId, application, wait)
```

Comment:
This was the problematic line: we should not be updating the recheck time when it has not yet expired, because we'd just keep pushing it forward into the future forever.